### PR TITLE
add mutual capacitance button wizard

### PR DESCRIPTION
**Pull request by canwail**
added Mutual capacitance Capacitive Touch Button.
Measurements are taken from [QTAN0079](http://ww1.microchip.com/downloads/en/appnotes/doc10752.pdf), Page 37.

------------

Thanks for creating a pull request to contribute to the KiCad footprint wizards! To speed up integration of your PR, please check the following items:

- [x] Provide a description for your PR (are you creating a new wizard or improving an existing one?)
- [x] Include a screenshot of the wizard screen with an example created footprint

Screenshot_748

### CODE REVIEW

**Comment by poeschlr** [comment]
I am not sure if this repo is the right place to contribute footprint wizards. We might need to ask over on the mailing list about that.

**Comment by sethhillbrand** [comment]
I think we'll clear out the old PRs here before moving over to GitLab.

File: mutualcap_button_wizard.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    def GetValue(self):
+        return "TouchButton-{width:g}x{height:g}mm".format(
+            width = pcbnew.ToMM(self.pads["Width"]),
+            height= pcbnew.ToMM(self.pads["Height"])
+            )
+
+    def GenerateParameterList(self):
+        #self.AddParam("Pads", "panel_thickness",self.uMM)
+        self.AddParam("Pads", "T (Panel thickness)", self.uMM, 1.6)
```
Comment (sethhillbrand) [inline mutualcap_button_wizard.py]: The values 'T' and 'Y width' are hard to parse.  I see they are used that way by Atmel, but let's get a better description.  Maybe "Surrounding Width"?

**Review by sethhillbrand — CHANGES_REQUESTED** [review]

File: mutualcap_button_wizard.py (hunk 1 above)
Comment (canwail) [reply mutualcap_button_wizard.py]: How about pt (panel thickness) and re and de for Drive/Receive electrode? Drive Electrode (currently X) is the thick surrounding one while Receive Electrode (currently Y) would be the inner thin one. 
Just calling it "Surrounding Width" i think would be a bit unclear as to what it is because the Drive Electrode "Finger" thickness also depends on that parameter.

Comment (sethhillbrand) [reply mutualcap_button_wizard.py]: I'd prefer to stay away from functional representations.  Instead, let's keep the graphical representations.  Maybe "Inner width" and "Outer width"?  Thickness tends to refer to the z-direction so is confusing in this context.

Comment (sethhillbrand) [reply mutualcap_button_wizard.py]: I see that inner is changed but there is still "Panel Thickness".  This is an uncommon term.  It would be good to rename it.  If you dislike "Outer width", you might call it "Frame Width".

Comment (canwail) [reply mutualcap_button_wizard.py]: I think there is a misunderstanding. The "Panel Thickness" refers to the thickness (z-direction) of whatever graphics panel or protection layer you put physically above the touchpad on the PCB (For example a acrylic sheet as graphical overlay or protection so that you dont touch the pcb directly). The "Outer width" is derived from the other parameters and can not be manually entered as a number.

Atmel calls it a front panel, Cypress (where this picture is from) calls it Overlay
article-2016december-simplifying-capacitive-touch-fig1

The internal variables were already changed to have i and o prefixes for inner and outer.

Comment (sethhillbrand) [reply mutualcap_button_wizard.py]: Thank you for that additional information.  I wasn't away of the relationship.

That being said, the border width or frame width is the same value as the overlay thickness in this recommended implementation.  A user would not be expected to know the relationship but they would be concerned about the width of the copper that is generated.  You could explain the relationship in the wizard description text but I'd prefer that the parameter be focused on the copper that is generated.

Comment (canwail) [reply mutualcap_button_wizard.py]: okay, parameter strings have been changed to inner and outer
